### PR TITLE
Retry validation on system failure

### PR DESCRIPTION
**Pull request by nzpr**
### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 1]
```diff
@@ -742,17 +746,37 @@ class RuntimeManagerImpl[F[_]: Concurrent: Metrics: Span: Log](
...
+          bondsPar =>
+            new IllegalArgumentException(
+              s"Incorrect number of results from query of current bonds in state ${PrettyPrinter
+                .buildString(hash)}: ${bondsPar.size}"
+            )
+        )(bondsPar => bondsPar.size == 1)
+        .map { bondsPar =>
+          toBondSeq(bondsPar.head)
+        }
+    )
```
Comment (tgrospic) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala:780]: Please add some TODO note that this is temp debugging solution.

**Review by tgrospic — APPROVED** [review]
LGTM

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala (hunk 1 above)
Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala:780]: https://github.com/rchain/rchain/pull/3176/commits/fc687229328b098d0394c087559d8eafda606595#diff-723ebc1ad7e9c98d566f4d96538fbb911b413c587f9b27e1ec4bc947efb34820R764

**Review by zsluedem — APPROVED** [review]
LGTM

**Comment by nzpr** [comment]
bors merge

**Pull request merged**